Rewrite getVsEntryRegInfo

s-perron opened the pull request:
The function getVsEntryRegInfo is a little long, and mixes a lot of word
together.  There is also a problem because on GFX9 the HW shader stage
does not match the user data registers as expected.  That is, a GS
shader uses the ES user data entries.

This PR will rewrite that code and fix up this problem.

amdvlk-admin commented:
### Test summary for commit 3e95913af11042d1c0b1a0697c5830c9b63b4737
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 1311c5cdfe6cde2faa349de66b2e09cbf49306aa</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5dbd470d8c1cbf8ee5d9a1204ba23a2fbf88a242</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 01f326c1e88cbf80e50a631ea0ce61a57152a7c0</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/227984)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31680/56996 (55.6%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 25316/56996 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31677/56996 (55.6%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 25319/56996 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32051/56996 (56.2%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 24945/56996 (43.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34932/56996 (61.3%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 22064/56996 (38.7%)</li>
</ul></details>


</ul></details>

kuhar commented on lgc/include/lgc/state/PalMetadata.h:
```diff
@@ -217,6 +218,19 @@ class PalMetadata {
   // The maximum possible value for the spill threshold entry in the PAL meatadata.
   static constexpr uint64_t MAX_SPILL_THRESHOLD = UINT_MAX;
 
+  unsigned int getUserDataCount(unsigned int callingConv);
```
nit: can use `unsigned` instead of mixing `unsigned` and `unsigned int`? Also everywhere else.

s-perron replied:
done

kuhar commented on lgc/state/PalMetadata.cpp:
```diff
@@ -719,94 +719,45 @@ void PalMetadata::eraseColorExportInfo() {
+unsigned PalMetadata::getUserDataCount(unsigned int callingConv) {
+  // Should we just define mmSPI_SHADER_PGM_RSRC2_* for all shader types? The comment before the definition of
+  // mmSPI_SHADER_PGM_RSRC2_VS in AbiMetadata.h says that do not want to.
+  static const std::map<unsigned, unsigned> callingConvToRsrc2Map = {
```
Do need these to be ordered? If not `llvm::SmallDenseMap` should perform better.

kuhar replied:
Also elsewhere

s-perron replied:
I looked into it, and I'm not sure how to cleanly initialize the value.  It is static, so I only want to do it once.  The initializer list does not work.  I will change it if there is a good way to do that.

kuhar replied:
Ah, that's odd that an initializer list constructor is missing. Can we use a simple helper function instead? Something like:
```
SmallDenseMap<unsigned, unsigned> build(std::initializer_list<std::pair<unsigned, unsigned>> init) {
  return {init.begin(), init.end()};
}
...
static SmallDenseMap m = build({{0, 1}}, {2, 3}});
```

And I will see if we can get a proper constructor added to LLVM and clean this up later.

s-perron replied:
I'll just leave it like this for now.  We can change it later if we want.

kuhar replied:
I added an initializer list constructor in https://github.com/llvm/llvm-project/commit/3c3165cfa0a4ff4f6c687b4ab4c141803d37009d

s-perron replied:
You are right.  The map is not  trivially destructible.  I'll have to change that.

s-perron replied:
I've replace them with an array of pairs, with is POD.

kuhar commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,179 @@ void PalMetadata::eraseFragmentInputInfo() {
+// =====================================================================================================================
+// Returns the calling convention of the first shader stage that will be executed in the pipeline.
+unsigned PalMetadata::getCallingConvenctionForFirstHardwareShaderStage() {
+  const unsigned numberOfShaderStages = 5;
```
nit: prefer `constexpr` for values that have to be known at compile time

s-perron replied:
done

kuhar commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,179 @@ void PalMetadata::eraseFragmentInputInfo() {
+// @param conv : The calling convention of the shader
+unsigned PalMetadata::getNumberOfSgprsBeforeUserData(unsigned int conv) {
+  // GFX9+ merged shader have an extra 8 SGPRs before user data.
+  if (conv != CallingConv::AMDGPU_VS && m_pipelineState->getTargetInfo().getGfxIpVersion().major >= 9)
```
nit: I realize this is an existing code, but I think the current convention is to compare using the `GfxIp` object, i.e., `....getGfxIpVersion() >= GfxIp{9, 0, 0}`

s-perron replied:
done

kuhar commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,179 @@ void PalMetadata::eraseFragmentInputInfo() {
+    if (reg >= lastReg)
+      return UINT_MAX;
+    if (static_cast<UserDataMapping>(userDataNode->second.getUInt()) == userDataMapping)
+      return (reg - firstReg);
```
nit: I'd drop the parens around subtraction

s-perron replied:
done

kuhar commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,179 @@ void PalMetadata::eraseFragmentInputInfo() {
+// @param callingConv : The calling convention of the shader stage
+//
+unsigned PalMetadata::getNumberOfSgprsAfterUserData(unsigned int callingConv) {
+  // Conservatively set the total number of input SGPRs. A merged shader with 8 SGPRs before user data
```
nit: SGPRs are inconsistently capitalized across the file

s-perron replied:
fixed.

kuhar commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,179 @@ void PalMetadata::eraseFragmentInputInfo() {
+}
+
+// =====================================================================================================================
+// Returns the offset of the  first user data register for the given calling convention.
```
nit: extra space before 'first`

s-perron replied:
done

kuhar commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,179 @@ void PalMetadata::eraseFragmentInputInfo() {
+//
+// @param callingConv : The calling convention of the shader stage
+//
+unsigned PalMetadata::getVertexIdOffset(unsigned int callingConv) {
```
Could we have the argument be `CallingConv` instead of unsigned? Just looking at the function signatures and the documentation, it's difficult to tell what calling convention is and how to grep for something related.

s-perron replied:
Unfortunately it cannot.  `CallingConv` is a namespace with an anonymous enum to enumerate the calling conventions.

kuhar commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,179 @@ void PalMetadata::eraseFragmentInputInfo() {
+
+  unsigned firstReg = firstUserDataNode->first.getUInt();
+  unsigned lastReg = firstReg + m_pipelineState->getTargetInfo().getGpuProperty().maxUserDataCount;
+  for (auto userDataNode = firstUserDataNode; userDataNode != m_registers.end(); ++userDataNode) {
```
nit: This could written as:
```
for (auto& userDataNode : llvm::make_range(firstUderDataNode, m_registers.end()) {
  unsigned reg = userdataNode.first.getUInt();
  ...
```

s-perron replied:
done

kuhar reviewed (commented):
Just some nits from me.

s-perron commented:
V2: Fixed the minor issues that Jakub found.

amdvlk-admin commented:
### Test summary for commit 0550e2e1c9298d83ff69882ccc607cffa421a0e2
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 1311c5cdfe6cde2faa349de66b2e09cbf49306aa</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5dbd470d8c1cbf8ee5d9a1204ba23a2fbf88a242</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 4f968af13313fd60f4a661943745d252ad14f2af</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/227984)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31680/56996 (55.6%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 25316/56996 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31677/56996 (55.6%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 25319/56996 (44.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32051/56996 (56.2%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 24945/56996 (43.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34932/56996 (61.3%)</li>
<li>Failed: 0/56996 (0.0%)</li>
<li>Not Supported: 22064/56996 (38.7%)</li>
</ul></details>


</ul></details>

trenouf commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,178 @@ void PalMetadata::eraseFragmentInputInfo() {
+                                                           {CallingConv::AMDGPU_ES, mmSPI_SHADER_USER_DATA_ES_0},
+                                                           {CallingConv::AMDGPU_GS, mmSPI_SHADER_USER_DATA_GS_0},
+                                                           {CallingConv::AMDGPU_VS, mmSPI_SHADER_USER_DATA_VS_0}};
+  static const std::map<unsigned, unsigned> shaderTableGfx9 = {{CallingConv::AMDGPU_LS, mmSPI_SHADER_USER_DATA_LS_0},
```
I guess you need this separate table/map because of merged es-gs using mmSPI_SHADER_USER_DATA_ES_0 on gfx9. (Was that wrong in my code? Is that how you ended up here as part of your GS work?) But I think the gfx9 table does not need entries for LS or ES at all.

Also same comment as before about using a static map.

s-perron replied:
Yes that is correct.  That was wrong in the original code, and I need to fix is before I can enable reloc geometry pipelines.  I've removed those entries.

s-perron replied:
I added them back in. I need to size of the two arrays to match make the assignment to the arrayref to work.

trenouf commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,178 @@ void PalMetadata::eraseFragmentInputInfo() {
+// @param conv : The calling convention of the shader
+unsigned PalMetadata::getNumberOfSgprsBeforeUserData(unsigned conv) {
+  // GFX9+ merged shader have an extra 8 SGPRs before user data.
+  if (conv != CallingConv::AMDGPU_VS && m_pipelineState->getTargetInfo().getGfxIpVersion() >= GfxIpVersion{9, 0, 0})
```
As this code now claims to be general and someone else might use it, perhaps it should work for CS (for which the answer should be 0), or at least assert that it is not being asked about CS.

trenouf commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,178 @@ void PalMetadata::eraseFragmentInputInfo() {
+//                            the first user data node for some shader stage.
+// @param userDataMapping : The user data mapping that is being search for.
+//
+unsigned PalMetadata::getOffsetOfUserDataReg(std::map<msgpack::DocNode, msgpack::DocNode>::iterator firstUserDataNode,
```
This now scans the user data regs in metadata once for each thing we want to find in it. Probably not much of an issue performance-wise; are you thinking it is more clear?

s-perron replied:
Yes, I did think it was clearer, and I did not think the performance will be significant.  The number of entries that are being searched is limited to 32 entries.

trenouf commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,178 @@ void PalMetadata::eraseFragmentInputInfo() {
+}
+
+// =====================================================================================================================
+// Returns the number of SGPRs that contain parameters for the shader after the user data.
```
Could the head-of-function comment mention that it is conservative not exact?

trenouf commented on lgc/state/PalMetadata.cpp:
```diff
@@ -969,3 +920,178 @@ void PalMetadata::eraseFragmentInputInfo() {
+  VGT_SHADER_STAGES_EN vgtShaderStagesEn;
+  vgtShaderStagesEn.u32All = m_registers[m_document->getNode(mmVGT_SHADER_STAGES_EN)].getUInt();
+  switch (callingConv) {
+  case CallingConv::AMDGPU_LS: // Before-GFX9 unmerged LS
```
No need for these two cases as this bit of code is reached only for gfx10+.

s-perron replied:
done.

trenouf reviewed (commented):
Generally looks good to refactor all of my splurgy code in that function, modulo a few niggles.

s-perron commented:
V3: Replaced static std::map variable with arrays.  Wrote function and types to accommodate that change.  Fixed up other small suggestion made by Tim.

s-perron commented:
V4: rebase

amdvlk-admin commented:
### Test summary for commit 5544533463a72b7b0276ce4881e1eb7d30998727
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 67d4b5e215558b039b69d940135835459dd91a51</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 846226b06470f9f4f9cfa52a3dfddd4c29fc95d4</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 53de8fec5020e6fec5411a795ce8c38fae6516ac</li>
</ul></details>

<details><summary>CTS tests (Failed: 1/216776)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32228/57047 (56.5%)</li>
<li>Failed: 0/57047 (0.0%)</li>
<li>Not Supported: 24819/57047 (43.5%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 29817/45635 (65.3%)</li>
<li><details><summary>Failed: 1/45635 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.subgroups.clustered.graphics.subgroupclusteredmul_f16vec2
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 15817/45635 (34.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32530/57047 (57.0%)</li>
<li>Failed: 0/57047 (0.0%)</li>
<li>Not Supported: 24517/57047 (43.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35505/57047 (62.2%)</li>
<li>Failed: 0/57047 (0.0%)</li>
<li>Not Supported: 21542/57047 (37.8%)</li>
</ul></details>


</ul></details>

s-perron commented:
V5: No change.  Modified the commit comment to get the tests to rerun.

s-perron commented:
V6: Modified the type of ArrayMap.  Guessing that will fix the error in the tests.  I cannot reproduce it locally, so I'm guessing what the problem is.

amdvlk-admin commented:
### Test summary for commit 82b8ecaeba89088daa3d19bc8f3bbafb9ca83fe0
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 67d4b5e215558b039b69d940135835459dd91a51</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 846226b06470f9f4f9cfa52a3dfddd4c29fc95d4</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 53de8fec5020e6fec5411a795ce8c38fae6516ac</li>
</ul></details>

<details><summary>CTS tests (Failed: 1/216464)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32228/57047 (56.5%)</li>
<li>Failed: 0/57047 (0.0%)</li>
<li>Not Supported: 24819/57047 (43.5%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 29523/45323 (65.1%)</li>
<li><details><summary>Failed: 1/45323 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.subgroups.arithmetic.compute.subgroupexclusivexor_u64vec2
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 15799/45323 (34.9%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32530/57047 (57.0%)</li>
<li>Failed: 0/57047 (0.0%)</li>
<li>Not Supported: 24517/57047 (43.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35505/57047 (62.2%)</li>
<li>Failed: 0/57047 (0.0%)</li>
<li>Not Supported: 21542/57047 (37.8%)</li>
</ul></details>


</ul></details>

amdvlk-admin commented:
### Test summary for commit 8b8607264efef937a8389997a89031f8f18484a1
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 67d4b5e215558b039b69d940135835459dd91a51</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 846226b06470f9f4f9cfa52a3dfddd4c29fc95d4</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 53de8fec5020e6fec5411a795ce8c38fae6516ac</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/228188)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32228/57047 (56.5%)</li>
<li>Failed: 0/57047 (0.0%)</li>
<li>Not Supported: 24819/57047 (43.5%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32226/57047 (56.5%)</li>
<li>Failed: 0/57047 (0.0%)</li>
<li>Not Supported: 24821/57047 (43.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32530/57047 (57.0%)</li>
<li>Failed: 0/57047 (0.0%)</li>
<li>Not Supported: 24517/57047 (43.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35505/57047 (62.2%)</li>
<li>Failed: 0/57047 (0.0%)</li>
<li>Not Supported: 21542/57047 (37.8%)</li>
</ul></details>


</ul></details>

s-perron commented:
@trenouf I believe I have addressed all of the comments.  Let me know if there is anything else.

trenouf reviewed (dismissed):
LGTM

amdrexu commented on lgc/state/PalMetadata.cpp:
```diff
@@ -49,6 +49,27 @@ using namespace lgc;
 
 namespace {
+
+struct KeyValuePair {
```
Add comments for this structure

amdrexu commented on lgc/state/PalMetadata.cpp:
```diff
@@ -980,3 +951,182 @@ bool PalMetadata::fragmentShaderUsesMappedBuiltInInputs() {
+// Returns to number of SGPRs before the SGPRs used for other purposes before the first user data SGPRs.
+//
+// @param conv : The calling convention of the shader
+unsigned PalMetadata::getNumberOfSgprsBeforeUserData(unsigned conv) {
```
conv -> callingConv

amdrexu commented on lgc/state/PalMetadata.cpp:
```diff
@@ -980,3 +951,182 @@ bool PalMetadata::fragmentShaderUsesMappedBuiltInInputs() {
+// Returns the upper bound on the number of SGPRs that contain parameters for the shader after the user data.
+//
+// @param callingConv : The calling convention of the shader stage
+//
```
This line is not needed

amdrexu commented on lgc/state/PalMetadata.cpp:
```diff
@@ -980,3 +951,182 @@ bool PalMetadata::fragmentShaderUsesMappedBuiltInInputs() {
+// Returns the offset of the vertex id in the parameter vertex registers.
+//
+// @param callingConv : The calling convention of the shader stage
+//
```
The same as above

amdrexu commented on lgc/state/PalMetadata.cpp:
```diff
@@ -980,3 +951,182 @@ bool PalMetadata::fragmentShaderUsesMappedBuiltInInputs() {
+// Returns the offset of the register containing the instance id in the shader.
+//
+// @param callingConv : The calling convention of the shader stage
+//
```
The same as above

amdrexu commented on lgc/state/PalMetadata.cpp:
```diff
@@ -980,3 +951,182 @@ bool PalMetadata::fragmentShaderUsesMappedBuiltInInputs() {
+// Returns true if the shader runs in wave32 mode.
+//
+// @param callingConv : The calling convention of the shader stage
+//
```
The same as above

amdrexu commented on lgc/state/PalMetadata.cpp:
```diff
@@ -980,3 +951,182 @@ bool PalMetadata::fragmentShaderUsesMappedBuiltInInputs() {
+// =====================================================================================================================
+// Returns the calling convention of the first hardware shader stage that will be executed in the pipeline.
+unsigned PalMetadata::getCallingConventionForFirstHardwareShaderStage() {
+  constexpr unsigned numberOfShaderStages = 5;
```
numberOfShaderStages -> hwShaderStageGfxCount

amdrexu commented on lgc/state/PalMetadata.cpp:
```diff
@@ -980,3 +951,182 @@ bool PalMetadata::fragmentShaderUsesMappedBuiltInInputs() {
 }
+
+// =====================================================================================================================
+// Returns the calling convention of the first hardware shader stage that will be executed in the pipeline.
```
I know the following functions are handling graphics shader stages. Do we need to add a assert like this assert(isGraphics())

s-perron replied:
I changed this function to handle CS.  Other functions already handle CS or will hit an llvm_unreachable.

JaxLinAMD commented:
@s-perron  Ping. You have unfinished conversation

JaxLinAMD commented:
> > @s-perron Ping. You have unfinished conversation
> 
> @JaxLinAMD FYI, Steven is on vacation until August 16.

oh, that's fine.

s-perron commented:
V7: Remove unnecessary empty comment lines.  Renamed a variable.  Added comment describing the new struct.  Updated `getCallingConventionForFirstHardwareShaderStage` and `getFirstUserDataReg` to handle compute shaders.  All in response to the review by @amdrexu.

amdvlk-admin commented:
### Test summary for commit e2a705fa4f7b1fd10837a823e2f2b92d52fa267e
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 0e83f232d8b70c6823f62bbd9dbf286a68037fe2</li>
<li>SPVGEN: dev a46945586e83a71cb07b11936eb5fe503afd6360</li>
<li>XGL: dev ec89b84f0b8a83aa2b29ed25cfd741c90f9e2688</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 3490d8b188fc398cc5467fed3e0bedd0dd8e61fb</li>
</ul></details>

<details><summary>CTS tests (Failed: 1/196384)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32108/57181 (56.2%)</li>
<li>Failed: 0/57181 (0.0%)</li>
<li>Not Supported: 25073/57181 (43.8%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 18284/24841 (73.6%)</li>
<li><details><summary>Failed: 1/24841 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.ubo.3_level_array.std140.column_major_mat4.both_comp_access
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 6556/24841 (26.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32488/57181 (56.8%)</li>
<li>Failed: 0/57181 (0.0%)</li>
<li>Not Supported: 24693/57181 (43.2%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35419/57181 (61.9%)</li>
<li>Failed: 0/57181 (0.0%)</li>
<li>Not Supported: 21762/57181 (38.1%)</li>
</ul></details>


</ul></details>

s-perron commented:
The failures seems to be the LGC failure that is unrelated to my change.

s-perron commented:
V8: No change just a rebase.
